### PR TITLE
Add blogpost "How to natively deploy Flink on Kubernetes with HA"

### PR DESCRIPTION
**Pull request by wangyang0918**
Add blogpost "How to natively deploy Flink on Kubernetes with HA"

Please use `./build.sh -f` to start a local web server for reviewing.

### CODE REVIEW

**Comment by wangyang0918** [comment]
cc @morsapaes, could you please help with reviewing this PR? After then we could pull in a committer to merge.

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (XComp) [inline _posts/2021-02-10-native-k8s-with-ha.md]:
```suggestion
Then, we’ll deep dive into Flink’s Kubernetes High Availability (HA) architecture and walk through a hands-on example of running a Flink [application cluster](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/#application-mode) on Kubernetes with HA enabled.
```

I suggest linking `application cluster` to the deployment docs. We rewrote the Deployment section recently. IMHO, it's a better entrypoint to do further reading.

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (XComp) [inline _posts/2021-02-10-native-k8s-with-ha.md:41]:
```suggestion
High Availability (HA) is a common requirement when bringing Flink to production: it helps to prevent a single point of failure for Flink clusters.
```

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (XComp) [inline _posts/2021-02-10-native-k8s-with-ha.md]:
```suggestion
        $ ./bin/flink run-application \
            --detached \
            --parallelism 4 \
            --target kubernetes-application \
            -Dkubernetes.cluster-id=k8s-ha-app-1 \
            -Dkubernetes.container.image=<user-image> \
            -Dkubernetes.jobmanager.cpu=0.5 \
            -Dkubernetes.taskmanager.cpu=0.5 \
            -Dtaskmanager.numberOfTaskSlots=4 \
            -Dkubernetes.rest-service.exposed.type=NodePort \
            -Dhigh-availability=org.apache.flink.kubernetes.highavailability.KubernetesHaServicesFactory \
            -Dhigh-availability.storageDir=s3://flink-bucket/flink-ha \
            -Drestart-strategy=fixed-delay \
            -Drestart-strategy.fixed-delay.attempts=10 \
            -Dcontainerized.master.env.ENABLE_BUILT_IN_PLUGINS=flink-s3-fs-hadoop-1.12.1.jar \
            -Dcontainerized.taskmanager.env.ENABLE_BUILT_IN_PLUGINS=flink-s3-fs-hadoop-1.12.1.jar \
            local:///opt/flink/usrlib/my-flink-job.jar
```

In the deployment docs we agreed on using long parameters. Another thing we tried to follow is to put one parameter per line. We hope that this improves the readability of the command. We could do that in the blog post as well, if you agree.

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (XComp) [inline _posts/2021-02-10-native-k8s-with-ha.md:57]:
```suggestion
docker built -t <user-image> .
docker push <user-image>
```
Should we add the docker commands here as well? This way, we would have a step by step list of commands that need to be called by the user. This way, we also make clear what `<user-image>` refers to in the next command.

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (XComp) [inline _posts/2021-02-10-native-k8s-with-ha.md]:
```suggestion
        ./bin/flink cancel --target kubernetes-application -Dkubernetes.cluster-id=<ClusterID> <JobID>
```
nit: Just for consistency purposes in case we want to go with the long parameter approach.

**Review by XComp — APPROVED** [review]
It's nice to read. Thanks for working on this @wangyang0918 and @morsapaes . I'm wondering whether we could add a paragraph referring to the [Generic Deployment](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/#deployment) and more specific [Native Kubernetes Deployment](https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/resource-providers/native_kubernetes.html) docs to have the documentation connected.

I have a few minor comments below

File: _posts/2021-02-10-native-k8s-with-ha.md
Comment (morsapaes) [reply _posts/2021-02-10-native-k8s-with-ha.md:41]: Mostly a matter of preference here. Both forms are correct, adding the "to" just makes it a tad bit more formal.

**Comment by morsapaes** [comment]
@wangyang0918 , pushed some changes after building to make the blogpost more organized. Also addressed @XComp 's suggestions.

The "Rebuild Page" commit should not be included in the PR. The committer will build the website after merging. 👌🏻

**Comment by rmetzger** [comment]
Thanks for the review @morsapaes and @XComp. Merging now

**Pull request closed without merging**
